src/connectors/hierarchical-menu/__tests__/connectHierarchicalMenu-test.js

```diff
@@ -809,120 +836,491 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/hierarchica
-    };
+      const widget = makeWidget({
+        attributes: ['categoriesLvl0', 'categoriesLvl1'],
+      });
 
-    describe('getWidgetState', () => {
-      test('should give back the object unmodified if there are no refinements', () => {
-        const [widget, helper] = getInitializedWidget();
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+      const actual = widget.getWidgetState(
+        {},
+        {
           searchParameters: helper.state,
-          helper,
-        });
+        }
+      );
 
-        expect(uiStateAfter).toBe(uiStateBefore);
+      expect(actual).toEqual({
+        hierarchicalMenu: {
+          categoriesLvl0: ['TopLevel', 'SubLevel'],
+        },
       });
+    });
 
-      test('should add an entry equal to the refinement', () => {
-        const [widget, helper] = getInitializedWidget();
-        helper.toggleRefinement('category', 'path');
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
-          searchParameters: helper.state,
-          helper,
-        });
+    test('returns the `uiState` without namespace overridden', () => {
+      // Uses the function getInitializedWidget once we've removed `getConfiguration`
+      const render = () => {};
+      const makeWidget = connectHierarchicalMenu(render);
+      const helper = jsHelper({}, '', {
+        hierarchicalFacets: [
+          {
+            name: 'categoriesLvl0',
+            attributes: ['categoriesLvl0', 'categoriesLvl1'],
+            separator: ' > ',
+            rootPath: null,
+            showParentLevel: true,
+          },
+        ],
+        hierarchicalFacetsRefinements: {
+          categoriesLvl0: ['TopLevelCategories > SubLevelCategories'],
+        },
+      });
 
-        expect(uiStateAfter).toMatchSnapshot();
+      const widget = makeWidget({
+        attributes: ['categoriesLvl0', 'categoriesLvl1'],
       });
 
-      test('should not overide other entries in the same namespace', () => {
-        const [widget, helper] = getInitializedWidget();
-        const uiStateBefore = {
+      const actual = widget.getWidgetState(
+        {
           hierarchicalMenu: {
-            otherCategory: ['path'],
+            countryLvl0: ['TopLevelCountry', 'SubLevelCountry'],
           },
-        };
-        helper.toggleRefinement('category', 'path');
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        },
+        {
           searchParameters: helper.state,
-          helper,
-        });
+        }
+      );
 
-        expect(uiStateAfter).toMatchSnapshot();
+      expect(actual).toEqual({
+        hierarchicalMenu: {
+          categoriesLvl0: ['TopLevelCategories', 'SubLevelCategories'],
+          countryLvl0: ['TopLevelCountry', 'SubLevelCountry'],
+        },
       });
+    });
+  });
+
+  describe('getWidgetSearchParameters', () => {
+    beforeEach(() => {
+      warning.cache = {};
```

Comment:
Can this be part of the tests setup?

Reply:
WDYM? Globally?

Reply:
I mean resetting the warning cache after each test so that they're independent.

Reply:
It's kinda what is happening no?

Reply:
I meant as part of our setup to no have to do it inside our test files.

Reply:
Yes so globally then.

Reply:
Yes, sure!